docs/docs/learn/shards/coding-with-shards.md

```diff
@@ -0,0 +1,220 @@
+
+| Variable Type   | Shard       | Alias  | Description                          |
+| :-------------- | :---------- | :----- | :----------------------------------- |
+| Local, Constant | `Ref`       | `=`    | Creates a local constant variable.   | 
+| Local, Mutable  | `Set`       | `>=`   | Creates a local mutable variable.    |
+| Global, Mutable | `Set`       | `>==`  | Creates a global mutable variable.   |
+| Mutable         | `Update`    | `>`    | Updates a mutable variable.          |
+
+In summary:
+
+- Use `=` to create **constant** variables.
+
+- Otherwise, use `>=` to create **local** variables, or `>==` to make them **global**.
+
+- Use `>` to update variable values.
+
+!!! note "Ligatures"
+    Some fonts will combine characters into a ligature. ">=" can appear as the ligature `>=`, while ">==" will appear as `>==`.
+
+## The shard
+
+A shard in its most basic code form consists of its name surrounded by brackets.
+
+![Some shard examples.](assets/shards-examples.png)
+
+The above example consists of 3 different predefined shards.
+
+Shards are named to make their purpose rather intuitive. `(Msg)` is the Message shard that prints a message to the console, while `(Math.Add)` is a Mathematics shard that add numbers together.
+
+!!! note
+    It is a good practice to name your code based on their purpose. This allows others to easily understand what your code achieves without getting too technical.
+
+    For example, the code `(Msg "Hello World!")` can be easily understood to be sending the message "Hello World" to the console. You do not need to delve into how `(Msg)` was coded to understand what it can do.
+
+
+A shard can take in an input, process that input, and produce an output. Shards also have *parameters* which behave as user-defined settings.
+
+For example, `Math.Add` has the parameter `Operand` which is defined by the user. It determines the value that will be added to the input. The final result is then produced as the output.
+
+![The parameters set for a shard affects it's behavior.](assets/shard-parameter.png)
+
+In code form, parameters are defined by the user within the brackets of the shard itself, after the shard's name. The above examples will appear as `5 (Math.Add 1)` and `5 (Math.Add 3)` in code. 
+ 
+Some shards have multiple parameters. When specifying values for multiple parameters, you will have to prepend your values with the parameter they are for.
+
+For example, the `Repeat` shard has four parameters: `Action`, `Times`, `Forever`, `Until`.
+
+We can utilize the `Repeat` shard with its different parameters as shown:
+
+=== "1 Parameter"
+
+    ```{.clojure .annotate linenums="1"}
+    (Repeat
+        (-> (Msg "Hello World!"))) ;; (1)(2)
+    ```
+
+    1. When no parameters are specified, the default parameter will be the first parameter the shard has. In this case, `Action` is the default parameter for `Repeat` and we set `(Msg "Hello World")` to it.
```

Comment:
It is not really a "default" parameter. It is just that when not specified by name, parameters are resolved in order.

So you could have the 4 parameters all "implied" if given in the right order. What explicitly naming parameters allows you to do is change the order, or skip some parameters (they will have their respective default value).